libraries/tock-cells/src/map_cell.rs

```diff
@@ -128,7 +115,8 @@ impl<T> MapCell<T> {
         if self.is_some() {
             self.occupied.set(false);
             let valref = unsafe { &mut *self.val.get() };
-            let res = closure(unsafe { &mut valref.some });
+            // TODO: change to .as_mut() once stabilized
```

Comment:
Could you add a link to the tracking issue here so this is easier to check up on?